### PR TITLE
Use flask error handlers

### PR DESCRIPTION
**Pull request by jpmartinspt**
## Done
Use flask error handlers to do the error handling...

## QA

- `./run`
- See it working

### CODE REVIEW

File: webapp/views.py [hunk 1]
```diff
@@ -226,23 +155,57 @@ def create_asset():
...
+    if asset:
+        if "width" not in asset.data and "width" in data:
+            asset.data["width"] = data["width"]
+
+        if "height" not in asset.data and "height" in data:
+            asset.data["height"] = data["height"]
+
+        db_session.commit()
+
+        error_message = "Asset already exists at {0}.".format(url_path)
```
Comment (tbille) [inline webapp/views.py]: "Asset already exists at {url_path}."

File: webapp/views.py [hunk 2]
```diff
@@ -226,23 +155,57 @@ def create_asset():
...
+
+    if asset:
+        if "width" not in asset.data and "width" in data:
+            asset.data["width"] = data["width"]
+
+        if "height" not in asset.data and "height" in data:
+            asset.data["height"] = data["height"]
+
+        db_session.commit()
+
```
Comment (tbille) [inline webapp/views.py]: all this doens't seem necessary since you are aborting

Comment (tbille) [reply webapp/views.py]: talked about it directly. Seems that it might be necessary.

File: webapp/views.py (hunk 1 above)
Comment (jpmartinspt) [reply webapp/views.py]: Changed here and in all other places where I was still using format.

File: webapp/views.py [hunk 3]
```diff
@@ -226,23 +155,56 @@ def create_asset():
...
+    # Save the file in Swift
+    file_manager.create(file_data, url_path)
+
+    # Save file info in Postgres
+    asset = Asset(file_path=url_path, data=data, created=datetime.utcnow())
+    db_session.add(asset)
+    db_session.commit()
+
+    return flask.jsonify(asset.as_json()), 201
```
Comment (tbille) [inline webapp/views.py]: you import `jsonify` so no need `flask`

Comment (jpmartinspt) [reply webapp/views.py]: Removed all `flask.{something}`

File: webapp/app.py [hunk 4]
```diff
@@ -34,15 +40,58 @@
 @app.errorhandler(401)
 @app.errorhandler(403)
 @app.errorhandler(404)
-def error_handler(exception=None):
-    code = getattr(exception, "code")
-    return {"status": f"{code}", "message": str(exception)}, code
+def error_handler(error=None):
+    code = getattr(error, "code")
+    return {"status": code, "message": str(error)}, code
```
Comment (nottrobin) [inline webapp/app.py]: If `code` (which is presumably an `int`) gets converted to a `str` automatically for passing to the context, why doesn't `error`?

Comment (jpmartinspt) [reply webapp/app.py]: `code` is not converted to a `str`, it shows in the JSON payload as a numeric type.

Comment (nottrobin) [reply webapp/app.py]: Oh I see! Fair enough.

**Pull request merged**